assembly.py

```diff
@@ -278,45 +284,42 @@ def assemble_trinity(
         First trim reads with trimmomatic, rmdup with prinseq,
         and random subsample to no more than 100k reads.
     '''
-    if outReads:
-        subsamp_bam = outReads
+
+    if already_preprocessed_bam:
+        subsamp_bam = inBam
+        read_stats = eval(util.file.slurp_file(preprocessed_bam_stats, noErr=True) or '(0,0,0,0,0,0)')
```

Comment:
`eval()` is generally considered [dangerous](https://nedbatchelder.com/blog/201206/eval_really_is_dangerous.html). Even if our input here is usually trusted, can we read in `preprocessed_bam_stats ` in a different way? If it is encoded in CSV/TSV format we can use the Python `csv` module, or else we could encode as JSON/YAML. Even parsing with a regular expression would be preferable to `eval()`